core/unit_test/headers_self_contained/CMakeLists.txt

```diff
@@ -10,7 +10,8 @@ file(GLOB KOKKOS_CONTAINERS_HEADERS RELATIVE ${BASE_DIR}/containers/src
 file(GLOB KOKKOS_ALGORITHMS_HEADERS RELATIVE  ${BASE_DIR}/algorithms/src
      ${BASE_DIR}/algorithms/src/*.hpp)
 
-if(NOT Kokkos_ENABLE_DEPRECATED_CODE_4)
+# erroring out when deprecated code is disaled and raising warnings that are treated as errors in the CI otherwise
+if(NOT Kokkos_ENABLE_DEPRECATED_CODE_4 OR Kokkos_ENABLE_DEPRECATION_WARNINGS)
```

Comment:
I opted for this form because mostly because I think it increases the chances we will remember to get rid of it when we remove all deprecated code in the future.